src/org/zaproxy/zap/spider/SpiderParam.java

```diff
@@ -696,8 +696,9 @@ public void setHandleParameters(HandleParametersOption handleParametersVisited)
 	 * {@code "USE_ALL"}, {@code "IGNORE_VALUE"}, {@code "IGNORE_COMPLETELY"}.
 	 * 
 	 * @param handleParametersVisited the new handle parameters visited value
+	 * @throws IllegalArgumentException if the given parameter is not a value of {@code HandleParametersOption}.
```

Comment:
That exception is not required to be declared. The call `valueOf(String)` might throw that exception (when there's no enum value that corresponds to the passed string).